onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -946,10 +946,37 @@ def aten_batch_norm_update_stats(
     raise NotImplementedError()
 
 
-def aten_bernoulli(self: TensorType, generator: Optional[str] = None) -> TensorType:
-    """bernoulli(Tensor self, *, Generator? generator=None) -> Tensor"""
+@torch_op("aten::bernoulli.default")
+def aten_bernoulli_default(self: TTensor) -> TTensor:
+    """Proximal implementation of aten::bernoulli.default
+
+    Other overloads under aten::bernoulli are
+      ['default', 'out', 'p', 'Tensor', 'Tensor_out', 'float_out'].
+    Note that due to the limitation of ONNX, we ignore the `generator` argument in
+      aten::bernoulli.default(Tensor self, *, Generator? generator=None) -> Tensor
+    """
+    rands = op.RandomUniformLike(
+        self,
+        high=1.0,
+        low=0.0,
+    )
+    output = op.Less(rands, self)
```

Comment:
May need CastLike(self, rands) for the comparison.

Reply:
No. It's generated by RandomUniform`Like`.

Reply:
RandomUniformLike only produces floating point outputs right? 

> T2 in ( tensor(double), tensor(float), tensor(float16) ): Constrain output types to float tensors.

When self is an integer, will there be any issues?